assets/src/dashboard/app/views/editorSettings/adNetwork/index.js

```diff
@@ -0,0 +1,196 @@
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import { useState, useEffect, useCallback } from 'react';
+import PropTypes from 'prop-types';
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import styled from 'styled-components';
+import {
+  FormContainer,
+  InlineForm,
+  InlineLink,
+  SettingForm,
+  SettingHeading,
+  VisuallyHiddenLabel,
+  TextInputHelperText,
+} from '../components';
+import { Dropdown } from '../../../../components';
+import { AD_NETWORK_TYPE, DROPDOWN_TYPES } from '../../../../constants';
+import { TranslateWithMarkup } from '../../../../../i18n';
+import { trackClick } from '../../../../../tracking';
+
+const AdNetworkDropdown = styled(Dropdown)`
+  & button {
+    border: ${({ theme, error }) =>
+      error
+        ? theme.DEPRECATED_THEME.borders.danger
+        : theme.DEPRECATED_THEME.borders.gray100};
+
+    &:active:enabled {
+      border: ${({ theme, error }) =>
+        error
+          ? theme.DEPRECATED_THEME.borders.danger
+          : theme.DEPRECATED_THEME.borders.action};
+    }
+  }
+`;
+const AdNetworkSettingForm = styled(SettingForm)`
+  padding-bottom: 0;
+`;
+
+export const TEXT = {
+  SECTION_HEADING: __('Monetization', 'web-stories'),
+  SLOT_ID_LABEL: __('Monetization type', 'web-stories'),
+  HELPER_MESSAGE_NONE: __(
+    'Monetize your content by showing ads in your Web Stories. <a>Learn more</a>.',
+    'web-stories'
+  ),
+  HELPER_MESSAGE_ADSENSE: __(
+    'Learn more about <a>how to monetize your Web Stories</a> using AdSense.',
+    'web-stories'
+  ),
+  HELPER_MESSAGE_ADMANAGER: __(
+    'Learn how to <a>enable programmatic demand in Web Stories</a> through Ad Manager.',
+    'web-stories'
+  ),
+  HELPER_LINK_NONE: __(
+    'https://amp.dev/documentation/guides-and-tutorials/develop/advertise_amp_stories/',
+    'web-stories'
+  ),
+  HELPER_LINK_ADSENSE: __(
+    'https://support.google.com/adsense/answer/10175505',
+    'web-stories'
+  ),
+  HELPER_LINK_ADMANAGER: __(
+    'https://support.google.com/admanager/answer/9416436',
+    'web-stories'
+  ),
+};
+
+const OPTIONS = [
+  {
+    label: __('None', 'web-stories'),
+    value: AD_NETWORK_TYPE.NONE,
+  },
+  {
+    label: __('Google AdSense', 'web-stories'),
+    value: AD_NETWORK_TYPE.ADSENSE,
+  },
+  {
+    label: __('Google Ad Manager', 'web-stories'),
+    value: AD_NETWORK_TYPE.ADMANAGER,
+  },
+];
+
+function AdNetworkSettings({ adNetwork: adNetworkRaw, handleUpdate }) {
+  const [adNetwork, setAdNetwork] = useState(adNetworkRaw);
+
+  useEffect(() => setAdNetwork(adNetworkRaw), [adNetworkRaw]);
+
+  let message;
+  let link;
+
+  if (AD_NETWORK_TYPE.ADMANAGER === adNetwork) {
+    message = TEXT.HELPER_MESSAGE_ADMANAGER;
+    link = TEXT.HELPER_LINK_ADMANAGER;
+  }
+
+  if (AD_NETWORK_TYPE.ADSENSE === adNetwork) {
+    message = TEXT.HELPER_MESSAGE_ADSENSE;
+    link = TEXT.HELPER_LINK_ADSENSE;
+  }
```

Comment:
nit: could memoize this so it only updates when `adNetwork` changes.

Reply:
Addressed now 👍